pymc/sampling/mcmc.py

```diff
@@ -446,6 +511,21 @@ def sample(
     initial_points = None
     step = assign_step_methods(model, step, methods=pm.STEP_METHODS, step_kwargs=kwargs)
 
```

Comment:
We should raise if we have a non-nuts stepper and a non-pymc nuts was requested.

Also tests